plugin/normalize.go

```diff
@@ -141,3 +142,109 @@ func SplitHostPort(s string) (host, port string, ipnet *net.IPNet, err error) {
 	}
 	return host, port, n, nil
 }
+
+// Subnets return a slice of prefixes with the desired mask subnetted from original network
+func Subnets(network *net.IPNet, newPrefixLen int) ([]net.IPNet) {
+	prefixLen, _ := network.Mask.Size()
+	maxSubnets := int(math.Exp2(float64(newPrefixLen)) / math.Exp2(float64(prefixLen)))
+	subnetsList := []net.IPNet{net.IPNet{network.IP, net.CIDRMask(newPrefixLen , 8*len(network.IP))}}
+	i := 1
+	for i < maxSubnets {
+		temp, _ := NextSubnet(&subnetsList[len(subnetsList)-1], newPrefixLen)
+		subnetsList = append(subnetsList, *temp)
+		i++
+	}
+
+	return subnetsList
+}
+// ClassfulFromCIDR return slice of "classful" (/8, /16, /24 or /32 only) CIDR's from any CIDR
+func ClassfulFromCIDR(s string) ([]string, error) {
+	_, n, err := net.ParseCIDR(s)
+
+	var networks []net.IPNet
+	var cidrs []string
+	if err == nil {
+		ones, _ := n.Mask.Size()
+		if ones % 8 == 0 {
+			cidrs = append(cidrs, n.String())
+		} else {
+			// TODO add RFC2317 / RFC4183 support for smaller than /24 subnets (if so need to ajust SplitHostPort to allow /32 subnets)
+			newMask := int(math.Ceil(float64(ones) / 8) ) * 8
+			networks = Subnets(n, newMask)
+		}
+		//cast to string
+		for i := 0; i < len(networks); i++ {
+			cidrs = append(cidrs, networks[i].String())
+		}
+	}
+	return cidrs, err
+}
+
+// NextSubnet returns the next available subnet of the desired mask size
+// starting for the maximum IP of the offset subnet
+// If the IP exceeds the maxium IP then the second return value is true
+func NextSubnet(network *net.IPNet, prefixLen int) (*net.IPNet, bool) {
```

Comment:
looks copied from https://github.com/apparentlymart/go-cidr/blob/master/cidr/cidr.go#L188

Reply:
Yes (added a note in the first commit) - seems like it drops after the change